node/flatpak-node-generator.py

```diff
@@ -1355,10 +1447,16 @@ class ProviderFactory:
     def create_lockfile_provider(self) -> LockfileProvider:
         raise NotImplementedError()
 
+    def create_rcfile_provider(self):
```

Comment:
Won't this crash on npm because there's no override?

Reply:
It will, `.npmrc` parsing is just not implemented yet.

Reply:
Implemented a common parser that should work for both `.yarnrc` and `.npmrc`, so npm in now supported.